File: server/medicore.layer/MediCore.Service/Model/HospitalImageModel.cs
using dna.core.service.Models;
using dna.core.service.Models.Abstract;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace MediCore.Service.Model
{
    public class HospitalImageModel : IModelBase
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Required")]
        public int HospitalId { get; set; }

        [Required(ErrorMessage = "Required")]
        public int ImageId { get; set; }

        
        public virtual HospitalModel Hospital { get; set; }

        public virtual ImageModel Image { get; set; }
    }
}

File: server/medicore.apps/src/model/hospitalImageModel.ts
import { Type } from 'class-transformer';
import { IEntityBase } from './IEntityBase';
import { HospitalModel, ImageModel } from '../model';

export class HospitalImageModel implements IEntityBase {
    id: number;
    hospitalId: number;
    imageId: number;
    @Type(() => HospitalModel)
    hospital: HospitalModel;
    @Type(() => ImageModel)
    image: ImageModel;

    constructor() {
        this.id = 0;
        this.hospitalId = 0;
        this.hospital = null;
        this.image = null;
    }
}

